From 82614eb704b420cbea74068889919248e58974d8 Mon Sep 17 00:00:00 2001
From: Alan Tull <atull@altera.com>
Date: Thu, 6 Mar 2014 16:22:27 -0600
Subject: [PATCH 031/172] FogBugz #99945-2: add gpio-dwapb driver to defconfig
 and dts

* Add the gpio-dwapb driver to the defconfig and device tree.
* Fix LED definitions in device tree
* Move the LED definitions that are specific to the Altera
  CycloneV socdk board to the proper device tree file.

Signed-off-by: Alan Tull <atull@altera.com>
[czou:Original patch taken from
https://github.com/altera-opensource/linux-socfpga.git socfpga-4.1]
Signed-off-by: czou <cao.zou@windriver.com>

v4: remove socfpga.dtsi changes which were upstreamed
    fix led definitions to match upstream socfpga.dtsi
    enable gpios in cyclone5 and arria5 dtsi
v3: fix arria5 socdk led definitions
    move cyclone5 led definitions to cyclone5 socdk dts
    move clock property to gpio main node, not gpio port nodes
v2: fix cyclone5 led definitions

Conflicts:

	arch/arm/boot/dts/socfpga_arria5.dtsi
	arch/arm/boot/dts/socfpga_cyclone5.dtsi
---
 arch/arm/boot/dts/socfpga_arria5_socdk.dts   | 21 +++++++++++++++
 arch/arm/boot/dts/socfpga_cyclone5.dtsi      |  6 -----
 arch/arm/boot/dts/socfpga_cyclone5_socdk.dts | 40 ++++++++++++++++++++++++++++
 arch/arm/configs/socfpga_defconfig           |  1 +
 4 files changed, 62 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boot/dts/socfpga_arria5_socdk.dts b/arch/arm/boot/dts/socfpga_arria5_socdk.dts
index ccaf417..4ce7440 100644
--- a/arch/arm/boot/dts/socfpga_arria5_socdk.dts
+++ b/arch/arm/boot/dts/socfpga_arria5_socdk.dts
@@ -60,8 +60,21 @@
 	rxc-skew-ps = <2000>;
 };
 
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&gpio2 {
+	status = "okay";
+};
+
 &i2c0 {
 	status = "okay";
+	speed-mode = <0>;
 
 	eeprom@51 {
 		compatible = "atmel,24c32";
@@ -80,6 +93,14 @@
 	vqmmc-supply = <&regulator_3_3v>;
 };
 
+&qspi {
+	status = "okay";
+};
+
+&uart0 {
+	status = "okay";
+};
+
 &usb1 {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/socfpga_cyclone5.dtsi b/arch/arm/boot/dts/socfpga_cyclone5.dtsi
index 06db951..a05e3df 100644
--- a/arch/arm/boot/dts/socfpga_cyclone5.dtsi
+++ b/arch/arm/boot/dts/socfpga_cyclone5.dtsi
@@ -38,12 +38,6 @@
 			cap-sd-highspeed;
 		};
 
-		ethernet@ff702000 {
-			phy-mode = "rgmii";
-			phy-addr = <0xffffffff>; /* probe for phy addr */
-			status = "okay";
-		};
-
 		sysmgr@ffd08000 {
 			cpu1-start-addr = <0xffd080c4>;
 		};
diff --git a/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts b/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts
index 258865d..744d30a 100644
--- a/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts
+++ b/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts
@@ -44,6 +44,29 @@
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
 	};
+
+	leds {
+		compatible = "gpio-leds";
+		hps0 {
+			label = "hps_led0";
+			gpios = <&portb 15 1>;
+		};
+
+		hps1 {
+			label = "hps_led1";
+			gpios = <&portb 14 1>;
+		};
+
+		hps2 {
+			label = "hps_led2";
+			gpios = <&portb 13 1>;
+		};
+
+		hps3 {
+			label = "hps_led3";
+			gpios = <&portb 12 1>;
+		};
+	};
 };
 
 &gmac1 {
@@ -60,12 +83,21 @@
 	rxc-skew-ps = <2000>;
 };
 
+&gpio0 {
+	status = "okay";
+};
+
 &gpio1 {
 	status = "okay";
 };
 
+&gpio2 {
+	status = "okay";
+};
+
 &i2c0 {
 	status = "okay";
+	speed-mode = <0>;
 
 	eeprom@51 {
 		compatible = "atmel,24c32";
@@ -85,6 +117,14 @@
 	vqmmc-supply = <&regulator_3_3v>;
 };
 
+&qspi {
+	status = "okay";
+};
+
+&uart0 {
+	status = "okay";
+};
+
 &usb1 {
 	status = "okay";
 };
diff --git a/arch/arm/configs/socfpga_defconfig b/arch/arm/configs/socfpga_defconfig
index e11e15a..5ceb660 100644
--- a/arch/arm/configs/socfpga_defconfig
+++ b/arch/arm/configs/socfpga_defconfig
@@ -84,6 +84,7 @@ CONFIG_I2C_DESIGNWARE_PLATFORM=y
 CONFIG_GPIOLIB=y
 CONFIG_GPIO_SYSFS=y
 CONFIG_GPIO_DWAPB=y
+CONFIG_GPIO_ALTERA=m
 CONFIG_PMBUS=y
 CONFIG_SENSORS_LTC2978=y
 CONFIG_SENSORS_LTC2978_REGULATOR=y
-- 
1.9.1

